README_INTERFACE_TESTS.md

```diff
@@ -0,0 +1,197 @@
+# Interface tests
+
+Charm relation interfaces are information exchange protocols for Juju charms.
+In this repository, under each interface specification, it is possible to store a list of interface tests. The interface maintainers will write and maintain those tests for the interfaces they own. 
+The goal of the interface tests is to determine whether a given charm satisfies an interface protocol.
+
+There are two personas that need to work with interface tests: interface authors and interface users.
+If you are an interface user, i.e. the owner of a charm that uses some of the interfaces specified in this repository, then you should look at the documentation in the [interface-tester-pytest](https://github.com/canonical/interface-tester-pytest) repository.
+
+If you are an interface author, this page is for you.
+
+# How-To write interface tests as an interface author
+
+Throughout this example we will write some interface tests for the `ingress` interface.
+This is the expected behaviour for the interface:
+1) on `ingress-relation-joined`, the requirer should write its part of the data to the relation. 
+2) on `ingress-relation-changed`, if the provider has written its part of the data to the relation, then the provider should publish its part of the data; otherwise it should publish no data.
+
+## A minimal test case
+
+Install the [interface tester package](https://github.com/canonical/interface-tester-pytest).:
+
+`pip install git+https://github.com/canonical/interface-tester-pytest`
+
+Specify the schema for the `ingress` interface databags in `./interfaces/ingress/v0/schema.py`.
+
+Create a python file at `<repo-root>/interfaces/ingress/v0/interface_tests/my_tests.py` with these contents:
+
+```python
+from scenario import State
+from interface_tester.interface_test import interface_test_case
+
+
+@interface_test_case(
+    event='ingress-relation-joined',
+    role='requirer',
+)
+def test_data_published_on_joined(output_state: State):
+    return
+```
+
+Run the test with `python run_matrix.py --include ingress`.
```

Comment:
Suggestion (sorry it's impossible to add suggestions with ```'s in them):

Run the test:
```
python run_matrix.py --include ingress
```